tests/Integration/Behaviour/Features/Context/Domain/Product/ProductCombinationFeatureContext.php

```diff
@@ -0,0 +1,136 @@
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\CombinationForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\QueryResult\ProductCombinationsForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductException;
+use Tests\Integration\Behaviour\Features\Context\Util\PrimitiveUtils;
+
+class ProductCombinationFeatureContext extends AbstractProductFeatureContext
+{
+    /**
+     * @When I generate combinations for product :productReference using following attributes:
+     *
+     * @param string $productReference
+     * @param TableNode $table
+     */
+    public function generateCombinations(string $productReference, TableNode $table): void
+    {
+        $tableData = $table->getRowsHash();
+        $groupedAttributeIds = $this->parseGroupedAttributeIds($tableData);
+
+        try {
+            $this->getCommandBus()->handle(new GenerateProductCombinationsCommand(
+                $this->getSharedStorage()->get($productReference),
+                $groupedAttributeIds
+            ));
+        } catch (ProductException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @Then product :productReference should have following combinations:
+     *
+     * @param string $productReference
+     * @param TableNode $table
+     */
+    public function assertProductCombinations(string $productReference, TableNode $table): void
+    {
+        $dataRows = $table->getColumnsHash();
+        $expectedCombinationsCount = count($dataRows);
+
+        /** @var ProductCombinationsForEditing $combinationsForEditing */
+        $combinationsForEditing = $this->getQueryBus()->handle(new GetProductCombinationsForEditing(
+            $this->getSharedStorage()->get($productReference),
+            $this->getDefaultLangId(),
+            $expectedCombinationsCount
+        ));
+
+        Assert::assertEquals(
+            count($combinationsForEditing->getCombinations()),
+            $expectedCombinationsCount,
+            'Unexpected combinations count'
+        );
+
+        /** @var CombinationForEditing $combinationForEditing */
+        foreach ($combinationsForEditing as $key => $combinationForEditing) {
+            Assert::assertEquals(
+                $dataRows[$key]['combination name'],
+                $combinationForEditing->getCombinationName(),
+                'Unexpected combination'
+            );
+        }
+    }
+
+    /**
+     * @param string $groupsAttributesReferencesPairs
+     *
+     * @return array
+     */
+    private function parseAttributes(string $groupsAttributesReferencesPairs): array
+    {
+        $attributeByGroupIds = [];
+        $groupAttributeReferencesPairs = explode(';', $groupsAttributesReferencesPairs);
+        foreach ($groupAttributeReferencesPairs as $groupAttributeReferencesPair) {
+            $groupAttributePair = explode(':', $groupAttributeReferencesPair);
```

Comment:
```suggestion
            list($groupReference, $attributeReference) = explode(':', $groupAttributeReferencesPair);
            $groupId = $this->getSharedStorage()->get($groupReference);
            $attributeId = $this->getSharedStorage()->get($attributeReference);
```
Just a suggestion, I find `list` pretty convenient when dealing with explode